Detectors/TPC/workflow/include/TPCWorkflow/TPCSectorCompletionPolicy.h

```diff
@@ -123,10 +135,17 @@ class TPCSectorCompletionPolicy
         }
       }
 
-      if (haveMatchedInput && activeSectors == validSectors.to_ulong()) {
+      // If the flag Config::RequireAll is set in the constructor arguments we require
+      // data from all inputs in addition to the sector matching condition
+      // To be fully correct we would need to require data from all inputs not going
+      // into the TPC policy, but that is not possible for the moment. That's why there is a possibly
+      // unhandled case if multiple TPC input routes are defined but a complete data set is coming over
+      // one of them. Not likely to be a use case, though.
+      if (haveMatchedInput && activeSectors == validSectors.to_ulong() &&
+          (!bRequireAll || nActiveInputRoutes == inputs.size())) {
         // we can process if there is input for all sectors, the required sectors are
         // transported as part of the sector header
-        op = CompletionPolicy::CompletionOp::Process;
+        op = CompletionPolicy::CompletionOp::Consume;
```

Comment:
@ktf I think I have been returning the wrong operation before, can you remind me about the difference between `Consume` and `Process`?

Reply:
In case an InputRecord is not complete, consume will invoke process and remove the entries from the record, process will process them but it will leave the record intact so that if more data arrives for the same timeslice, the old entries in the record will also still be there.

Reply:
ah, thanks